jennchenn/uwexplore
# Implemented course conflict flagging

## Implementation description

- Updated Calendar component to work with courses array/objects
- Changed the start and end times of each class to be in ms, instead of minutes, so I could get the start/end time of each class easily
- Added red warning icon to the bottom left corner of classes that have a conflict 
- Added a short description of which class is the conflict when opening the modal (after clicking on a course)
- One big func to extract each individual class as an object, and put them in an array
- One big func to determine overlapping class times
- Updated SearchCards.tsx to also use ms for calculating the class times
- Added comments where I thought appropriate
- Im not optimizing anything at this point lol, so if there are any suggestions lmk



## Steps to test

Verify:
1. Do all conflicting classes have a red triangle warning icon?
2. When clicking on one of these courses, does the modal list the courses it is conflicting with?
3. Can the background colour of courses still be changed?
4. (option) cross-verify info from the search cards to the popup module



## What should reviewers focus on?

- Do the comments in the code make sense
- I see some titles are not showing, this will need to be fixed later during the styles update
